### ISSUE TITLE
'ignore_label' query

### ISSUE THREAD

**Issue by msinto93** [DrSleep/tensorflow-deeplab-resnet#168]
Hi, I am confused about the use of the ignore_label. As I understand it this is used to exclude the void/unlabeled classes (and padded 0s from ImageReader stuff) from contributing to the loss calculation. Yet I see nothing in the code where this ignore_label is provided to the loss or gradient calculations to explicitly tell it to ignore these labels. Am I missing something or is it merely enough to set it to a high value (255) to have it 'ignored'?

**Comment by zzllc3** [comment]
I have the same confuse!
I don't understand the  meaning of the function "random_crop_and_pad_image_and_lables" in "image_reader.py".
Maybe we can get better result without using this operation.

**Comment by DrSleep** [comment]
everything higher than the number of classes is ignored:
https://github.com/DrSleep/tensorflow-deeplab-resnet/blob/master/train.py#L166

**Comment by zzllc3** [comment]
@DrSleep thanks a lot for your reply,my image-groundtruth-prediction looks like this,do you know the reason?Thanks 
![1q](https://user-images.githubusercontent.com/22652597/37810228-e50a57ea-2e8d-11e8-92b1-01898dc45501.jpg)

**Comment by zzllc3** [comment]
@DrSleep  thanks a lot for your reply,my image(with padding
![1q](https://user-images.githubusercontent.com/22652597/37810256-0616351c-2e8e-11e8-8426-b81fa23300f1.jpg)
)-groundtruth-prediction looks like this,do you know the reason?Thanks

**Comment by zzllc3** [comment]
Sorry to bother you.I guess i know the reason,because after subtracting the IMG_MEAN ,the pixel of target are close to the pixel of padding part.It won't influence the training processing and the result.
